src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2143,13 +2157,20 @@ private void ReportNullabilityMismatchWithTargetDelegate(SyntaxNode syntax, Name
             }
         }
 
-        private static TypeSymbolWithAnnotations InferResultNullability(BoundExpression operand, Conversion conversion, TypeSymbol targetType, TypeSymbolWithAnnotations operandType, bool fromConversionNode = false)
+        private TypeSymbolWithAnnotations InferResultNullability(BoundExpression operand, Conversion conversion, TypeSymbol targetType, TypeSymbolWithAnnotations operandType, bool fromConversionNode = false, bool fromExplicitConversion = false)
```

Comment:
>InferResultNullability [](start = 42, length = 22)

I think the method should be renamed to reflect the change in what it is doing. For example, I wouldn't expect it to report any diagnostics. Also, it would be good to make it clear what kind of diagnostics it is responsible to report.